homeassistant/components/upnp/sensor.py

```diff
@@ -1,273 +1,243 @@
-
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    async def async_update(self):
-        """Get the latest information from the IGD."""
-        if self._type_name == BYTES_RECEIVED:
-            self._state = await self._device.async_get_total_bytes_received()
-        elif self._type_name == BYTES_SENT:
-            self._state = await self._device.async_get_total_bytes_sent()
-        elif self._type_name == PACKETS_RECEIVED:
-            self._state = await self._device.async_get_total_packets_received()
-        elif self._type_name == PACKETS_SENT:
-            self._state = await self._device.async_get_total_packets_sent()
-
-
-class PerSecondUPnPIGDSensor(UpnpSensor):
-    """Abstract representation of a X Sent/Received per second sensor."""
-
-    def __init__(self, device, direction):
-        """Initialize sensor."""
-        super().__init__(device)
-        self._direction = direction
-
-        self._state = None
-        self._last_value = None
-        self._last_update_time = None
-
-    @property
-    def unit(self) -> str:
-        """Get unit we are measuring in."""
-        raise NotImplementedError()
-
-    async def _async_fetch_value(self):
-        """Fetch a value from the IGD."""
-        raise NotImplementedError()
-
-    @property
-    def unique_id(self) -> str:
-        """Return an unique ID."""
-        return f"{self._device.udn}_{self.unit}/sec_{self._direction}"
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return self._coordinator.last_update_success
 
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return f"{self._device.name} {self.unit}/sec {self._direction}"
+        return f"{self._device.name} {self._sensor_type['name']}"
 
     @property
-    def icon(self) -> str:
-        """Icon to use in the frontend, if any."""
-        return "mdi:server-network"
+    def unique_id(self) -> str:
+        """Return an unique ID."""
+        return f"{self._device.name}_{self._device.udn}_{self._sensor_type['name']}"
 
     @property
     def unit_of_measurement(self) -> str:
         """Return the unit of measurement of this entity, if any."""
-        return f"{self.unit}/{TIME_SECONDS}"
+        return self._sensor_type["unit"]
 
-    def _is_overflowed(self, new_value) -> bool:
-        """Check if value has overflowed."""
-        return new_value < self._last_value
-
-    async def async_update(self):
-        """Get the latest information from the UPnP/IGD."""
-        new_value = await self._async_fetch_value()
-
-        if self._last_value is None:
-            self._last_value = new_value
-            self._last_update_time = dt_util.utcnow()
-            return
+    @property
+    def device_info(self) -> Mapping[str, any]:
+        """Get device info."""
+        return {
+            "connections": {(dr.CONNECTION_UPNP, self._device.udn)},
+            "identifiers": {(DOMAIN, self._device.udn)},
```

Comment:
No need to put both a connection and an identifier in. They are treated the same, so just use the connection one. 
```suggestion
            "connections": {(dr.CONNECTION_UPNP, self._device.udn)},
```